### PR TITLE
[HOPSWORKS-909] Upgrade Kafka to 2.3.0

### PR DESCRIPTION
**Pull request by kai-chi**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-909

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
upgrade Kafka to version 2.3

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hopsworks-common/pom.xml [hunk 1]
```diff
@@ -109,7 +109,7 @@
     <dependency>
       <groupId>org.apache.kafka</groupId>
       <artifactId>kafka-clients</artifactId>
-      <version>1.1.0</version>
+      <version>2.3.0</version>
```
Comment (tkakantousis) [inline hopsworks-common/pom.xml]: @kai-chi can you check if this new version brings new dependencies, such as new jackson version, in Hopsworks?

Comment (kai-chi) [reply hopsworks-common/pom.xml]: yes, it takes jackson from 2.9.4 to 2.9.9 but no more

Comment (tkakantousis) [reply hopsworks-common/pom.xml]: @kai-chi Can you do a mvn dependency tree check to see if Hopsworks has any other jackson dependencies? We need to minimize the risk of having conflicting versions. 
Also, do we use any code `kafka-clients` methods in Hopsworks that would make use of jackson? Probably not, so if that's the case we could potentially exclude jackson.

Comment (kai-chi) [reply hopsworks-common/pom.xml]: I added exclusions to kafka dependency. the mvn dependency tree shows that:
- kafka-clients introduces new dependency com.github.luben:zstd-jni:jar:1.4.0-1 and takes org.lz4:lz4-java from 1.4 to 1.6.0
- kafka takes jackson-databind from 2.9.4 to 2.9.9 and doesn't introduce anything

Comment (tkakantousis) [reply hopsworks-common/pom.xml]: Great, then we can merge this after we test that Hopsworks code using kafka and kafka-clients is not affected by that

Comment (kai-chi) [reply hopsworks-common/pom.xml]: tests passed

**Pull request merged**